service/history/handler.go

```diff
@@ -111,10 +111,8 @@ func NewHandler(
 		status:          common.DaemonStatusInitialized,
 		config:          config,
 		tokenSerializer: common.NewProtoTaskTokenSerializer(),
-		rateLimiter: quotas.NewDynamicRateLimiter(
-			func() float64 {
-				return float64(config.RPS())
-			},
+		rateLimiter: quotas.NewDefaultIncomingDynamicRateLimiter(
+			func() float64 { return float64(config.RPS()) },
```

Comment:
Conceptual question: why do history and matching need throttler at all? Shouldn't throttling to  be handled on FE only?

Reply:
each individual host / service need to do proper self protection (according to the current resource usage)

Reply:
But how can produce such load besides FE? Yes there is matching <-> history communication, but they are not user facing and probably shouldn't be throttled.

Reply:
It would be great to add metric to see if history and matching throttles actually throttle.

Reply:
> but they are not user facing and probably shouldn't be throttled.

then the host can be overloaded